sources/api/schnauzer/src/helpers.rs

```diff
@@ -1416,8 +1488,8 @@ pub fn oci_defaults(
     let section =
         OciSpecSection::from_str(oci_spec_section).context(error::InvalidOciSpecSectionSnafu)?;
     let result_lines = match section {
-        OciSpecSection::Capabilities => oci_spec_capabilities(oci_defaults_values)?,
-        OciSpecSection::ResourceLimits => oci_spec_resource_limits(oci_defaults_values)?,
+        OciSpecSection::Capabilities => oci_spec_capabilities(oci_defaults_values, runtime)?,
+        OciSpecSection::ResourceLimits => oci_spec_resource_limits(oci_defaults_values, runtime)?,
```

Comment:
This seems like a lot of indirection; consider just calling the associated method directly. That way all of the output is generated in the same place.

```suggestion
        (Runtime::Docker(r), OciSpecSection::Capabilities) => r.capabilities(oci_defaults_values)?,
        (Runtime::Docker(r), OciSpecSection::ResourceLimits) => r.resource_limits(oci_defaults_values)?,
        (Runtime::Containerd(r), OciSpecSection::Capabilities) => r.capabilities(oci_defaults_values)?,
        (Runtime::Containerd(r), OciSpecSection::ResourceLimits) => r.resource_limits(oci_defaults_values)?,
```

Might also be better to do this part before the call so the methods get the preferred form of data:
```
let oci_default_rlimits: HashMap<OciDefaultsResourceLimitType, OciDefaultsResourceLimit> = ...;
let oci_default_capabilities: HashMap<OciDefaultsCapability, bool> = ...;
```

Reply:
Implemented Runtime having get_capabilities and get_resource_limits as serde_plain complains about deserializing to enum as:
enum {
Docker(Docker),
Containerd(Containerd)
}